### PR TITLE
docs: Add additional CMake documentation

### PR DESCRIPTION
**Pull request by chuckatkins**
Add some additional documentation for how to integrate this into another CMake project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,10 @@
 
 - [Design goals](#design-goals)
 - [Integration](#integration)
+  - [CMake](#cmake)
+    - [External](#external)
```
Comment (nlohmann) [inline README.md]: The table of contents seems to be rather crowded.

- Please leave "External", "Embedded", and "Supporting Both" out of the table of contents.
- Please add "Package Managers" to the table of contents.

**Comment by chuckatkins** [comment]
@nlohmann TOC fixed and cleaned up.

**Comment by chuckatkins** [comment]
Looks like Travis got a little borked.  I'll kick the branch to cycle the CI again.

**Comment by chuckatkins** [comment]
@nlohmann this should be good to go now to once CI finishes.

**Comment by nlohmann** [comment]
Thanks!

**Pull request merged**